dht/test/common.go

```diff
@@ -157,16 +157,23 @@ func NewDHTNode(ctx context.Context, runenv *runtime.RunEnv, opts *SetupOpts, id
 		return nil, nil, err
 	}
 
+	cm, err := connmgr.NewConnManager(min, max, connmgr.WithGracePeriod(ConnManagerGracePeriod))
+	if err != nil {
+		return nil, nil, err
+	}
+
 	libp2pOpts := []libp2p.Option{
 		libp2p.Identity(idKey),
 		// Use only the TCP transport without reuseport.
-		libp2p.Transport(func(u *tptu.Upgrader) *tcp.TcpTransport {
-			tpt := tcp.NewTCPTransport(u)
-			tpt.DisableReuseport = true
+		libp2p.Transport(func(u transport.Upgrader) *tcp.TcpTransport {
```

Comment:
That's not how you constructor a libp2p transport. You just pass in the constructor, without executing it.

Reply:
much cleaner, updated!